cmd/gpu_fakedev/gpu_fakedev.go

```diff
@@ -270,6 +273,8 @@ func generateDriFiles(opts genOptions) {
 	log.Printf("Generating fake DRI device(s) sysfs, debugfs and devfs content under '%s' & '%s'",
 		sysfsPath, devfsPath)
 
+	makeXelinkSideCar(opts.Capabilities["connection-topology"], opts.DevCount, opts.TilesPerDev, opts.Capabilities["connections"])
```

Comment:
Move this call after the loop, so that it's not between logging of the loop start and the loop itself.

It would also be better to extract the values from `opts` in `makeXelinkSideCar()`, then there's no chance of giving int args in incorrect order and it's more consistent with `opts` usage in rest of this function.

Reply:
Yes makeXelinkSideCar() can be last call in func. It was in that position because testing was easier with using sudo.
Options processing are now also moved makeXelinkSideCar, opts variable need to be forward with makeXelinkSideCar(opts) call.